mlfinlab/labeling/labeling.py

```diff
@@ -0,0 +1,288 @@
+    (tao ≪ sigma_t_i,0 ), considering the prevailing volatility. Snippet 3.1 computes the daily volatility
+    at intraday estimation points, applying a span of lookback days to an exponentially weighted moving
+    standard deviation.
+
+    See the pandas documentation for details on the pandas.Series.ewm function.
+
+    Note: This function is used to compute dynamic thresholds for profit taking and stop loss limits.
+
+    :param close: Closing prices
+    :param lookback: lookback period to compute volatility
+    :return: series of daily volatility value
+    """
+    # daily vol re-indexed to close
+    df0 = close.index.searchsorted(close.index - pd.Timedelta(days=1))
+    df0 = df0[df0 > 0]
+    df0 = (pd.Series(close.index[df0 - 1], index=close.index[close.shape[0] - df0.shape[0]:]))
+
+    df0 = close.loc[df0.index] / close.loc[df0.values].values - 1  # daily returns
+    df0 = df0.ewm(span=lookback).std()
+    return df0
+
+
+# Snippet 3.2, page 45, Triple Barrier Labeling Method
+def apply_pt_sl_on_t1(close, events, pt_sl, molecule):  # pragma: no cover
+    """
+    Snippet 3.2, page 45, Triple Barrier Labeling Method
+
+    This function applies the triple-barrier labeling method. It works on a set of
+    datetime index values (molecule). This allows the program to parallelize the processing.
+
+    :param close: (series) close prices
+    :param events: (series) of indices that signify "events" (see get_t_events function
+    for more details)
+    :param pt_sl: (array) element 0, indicates the profit taking level; element 1 is stop loss level
+    :param molecule: (an array) a set of datetime index values for processing
+    :return: Dataframe of timestamps of when first barrier was touched
+    """
+    # Apply stop loss/profit taking, if it takes place before t1 (end of event)
+    events_ = events.loc[molecule]
+    out = events_[['t1']].copy(deep=True)
+
+    profit_taking_multiple = pt_sl[0]
+    stop_loss_multiple = pt_sl[1]
+
+    # Profit taking active
+    if profit_taking_multiple > 0:
+        profit_taking = profit_taking_multiple * events_['trgt']
+    else:
+        profit_taking = pd.Series(index=events.index)  # NaNs
+
+    # Stop loss active
+    if stop_loss_multiple > 0:
+        stop_loss = -stop_loss_multiple * events_['trgt']
+    else:
+        stop_loss = pd.Series(index=events.index)  # NaNs
+
+    # Get events
+    for loc, vertical_barrier in events_['t1'].fillna(close.index[-1]).iteritems():
+        df0 = close[loc: vertical_barrier]  # path prices
+        df0 = (df0 / close[loc] - 1) * events_.at[loc, 'side']  # path returns
+        out.loc[loc, 'sl'] = df0[df0 < stop_loss[loc]].index.min()  # earliest stop loss
+        out.loc[loc, 'pt'] = df0[df0 > profit_taking[loc]].index.min()  # earliest profit taking
+
+    return out
+
+
+# Snippet 3.4 page 49, Adding a Vertical Barrier
+def add_vertical_barrier(t_events, close, num_days=1):
```

Comment:
What if vertical barrier is not days, but seconds, hours? Can we make it more universal, maybe number of bars?

Reply:
I like this idea.

Lets rather create a new issue and label it as a good first issue. That way we are aware that we need to add this functionality and its a good ticket to have new people get started on. 

We should probably create a few different functions, one that is time based and then next bar count based.

Reply:
I've created an issue for that